pkg/resources/openstack/dns.go

```diff
@@ -30,15 +30,17 @@ const (
 	typeDNSRecord = "dNSRecord"
 )
 
-var listDNSFunctions = []listFn{
-	listDNSRecordsets,
-}
-
-func listDNSRecordsets(cloud openstack.OpenstackCloud, clusterName string) ([]*resources.Resource, error) {
+func (os *clusterDiscoveryOS) ListDNSRecordsets() ([]*resources.Resource, error) {
```

Comment:
Just a nit (no need to fix) - because this is usually referred to either within the package where it's contextually obvious, or externally as `openstack.ClusterDiscoveryOS` (if it was exported), it's usually fine to drop the `OS` or something like `Openstack`.  When it's `openstack.OpenstackClusterDiscovery` this is often nicknamed "stuttering".

And yes, I'm very guilty of it :-)